src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -74,16 +74,26 @@ public static function setPlaceholder(string $placeholder, array $values): void
     }
 
     /**
-     * Sets a common prefix for dynamic placeholder values.
+     * Adds a common prefix for dynamic placeholder values.
      *
      * Matching configuration values will be skipped from being processed and are returned as is, thus preserving the
      * placeholder. An exact match provided by {@see setPlaceholder()} might take precedence.
      *
      * @internal
      */
-    public static function setPlaceholderUniquePrefix(string $prefix): void
+    public static function addPlaceholderUniquePrefix(string $prefix): void
     {
-        self::$placeholderUniquePrefix = $prefix;
+        self::$placeholderUniquePrefixes[] = $prefix;
+    }
+
+    /**
+     * Resets all current placeholder unique prefixes available.
+     *
+     * @internal
+     */
+    public static function resetPlaceholderUniquePrefixes(): void
```

Comment:
i tend to prefer to keep a single `resetPlaceholders()`, as 'prefixed placeholder' are still placeholders.

Reply:
MergeExtensionConfigurationPass only uses placeholder unique prefixes and ValidateEnvPlaceholdersPass only uses direct placeholders, that's why I splitted.

Reply:
i see. It was never intended to e.g. reset direct placeholders, while keeping prefixes. Or vice versa.

on setup; configure placeholders, either direct or by prefix
on teardown: reset placeholders (either direct or by prefix)

Reply:
Alright that's fine for me, I'll change it :)